src/github-release.ts

```diff
@@ -69,67 +72,99 @@ export class GitHubRelease {
   }
 
   async createRelease(): Promise<ReleaseCreateResponse | undefined> {
-    const gitHubReleasePR:
-      | GitHubReleasePR
-      | undefined = await this.gh.findMergedReleasePR(this.labels);
-    if (gitHubReleasePR) {
-      const version = `v${gitHubReleasePR.version}`;
-
-      checkpoint(
-        `found release branch ${chalk.green(version)} at ${chalk.green(
-          gitHubReleasePR.sha
-        )}`,
-        CheckpointType.Success
-      );
-
-      const changelogContents = (
-        await this.gh.getFileContents(this.addPath(this.changelogPath))
-      ).parsedContent;
-      const latestReleaseNotes = GitHubRelease.extractLatestReleaseNotes(
-        changelogContents,
-        // For monorepo releases, the library name is prepended to the tag and branch:
-        version.split('-').pop() || version
-      );
-      checkpoint(
-        `found release notes: \n---\n${chalk.grey(latestReleaseNotes)}\n---\n`,
-        CheckpointType.Success
-      );
-
-      // Attempt to lookup the package name from a well known location, such
-      // as package.json, if none is provided:
-      if (this.packageName === undefined && this.releaseType) {
-        this.packageName = await ReleasePRFactory.class(
-          this.releaseType
-        ).lookupPackageName(this.gh);
-      }
-      if (this.packageName === undefined) {
-        throw Error('could not determine package name for release');
-      }
-
-      const release = await this.gh.createRelease(
-        this.packageName,
-        version,
-        gitHubReleasePR.sha,
-        latestReleaseNotes
-      );
-      // Add a label indicating that a release has been created on GitHub,
-      // but a publication has not yet occurred.
-      await this.gh.addLabels([GITHUB_RELEASE_LABEL], gitHubReleasePR.number);
-      // Remove 'autorelease: pending' which indicates a GitHub release
-      // has not yet been created.
-      await this.gh.removeLabels(this.labels, gitHubReleasePR.number);
-      return release;
+    let gitHubReleasePRs: GitHubReleasePR[] = [];
+    const rootPath = this.path;
+    const releases = [];
+    if (this.monorepoTags) {
+      gitHubReleasePRs = await this.gh.findMergedReleasePRs(this.labels, 25);
     } else {
+      const releasePR = await this.gh.findMergedReleasePR(this.labels, 25);
```

Comment:
We run the `github-release` step immediately after a PR is merged, so I figured that a page size of `100` should be far more than we need.

`25` leaves a bit of a buffer, in case something odd happens, e.g., we miss running release-please for a couple merged PRs, but in practice this could probably be a lower number.

Reply:
left a comment.